src/sinks/statsd.rs

```diff
@@ -0,0 +1,351 @@
+use crate::{
+    buffers::Acker,
+    event::{metric::Direction, Event, Metric},
+    sinks::util::{BatchServiceSink, Buffer, SinkExt},
+    topology::config::{DataType, SinkConfig},
+};
+use futures::{future, sink::Sink, Future, Poll};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, UdpSocket};
+use std::time::Duration;
+use tower::{Service, ServiceBuilder};
+
+pub struct StatsdSvc {
+    client: Client,
+}
+
+pub struct Client {
+    socket: UdpSocket,
+    address: SocketAddr,
+}
+
+impl Client {
+    pub fn new(address: SocketAddr) -> Self {
+        let from = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0);
+        let socket = UdpSocket::bind(&from).expect("failed to bind to udp listener socket");
```

Comment:
Is it safe to assume we won't see this error?

Reply:
I think there is a small chance. Do you think we need to add some error handling?

Reply:
Seems relevant #263

Reply:
Yes, looks like what you added handles this the way I'd expect.